.github/scripts/generate_binary_build_matrix.py

```diff
@@ -51,7 +51,7 @@
         "nvidia-curand-cu11==10.3.0.86; platform_system == 'Linux' and platform_machine == 'x86_64' | "
         "nvidia-cusolver-cu11==11.4.1.48; platform_system == 'Linux' and platform_machine == 'x86_64' | "
         "nvidia-cusparse-cu11==11.7.5.86; platform_system == 'Linux' and platform_machine == 'x86_64' | "
-        "nvidia-nccl-cu11==2.21.5; platform_system == 'Linux' and platform_machine == 'x86_64' | "
```

Comment:
Here https://pypi.org/project/nvidia-nccl-cu11/ version: 2.21.5 is the latest one. I guess we only would need to update cu 12.1 and 12.4

Reply:
@eqy Do you know why there are no binaries released for 11.8 here? Is this intentional or a release bug?

Reply:
@atalman Unfortunately, this isn't easily possible as Conda use a statically linked one which builds against the git submodule. We could build that one 2.22.5, but it's untested and violates an testing assert. @malfet What are the thoughts going for now that 11.8 is not supported by Nvidia anymore?

Reply:
If 2.22.3 is not supported for 11.8 and we need 2.22.3, we should stop supporting 11.8, but IMO this is a pretty big step to make... On the other hand, those constrains are only needed for poetry, aren't they? In that case, we might build/link with nccl statically for 11.8, but dynamically for 12.x

Reply:
We kinda decided to stop supporting 11.8, so maybe we could update nccl? cc @kwen2501